ffconv/file_processor.py

```diff
@@ -1,36 +1,12 @@
+"""
+This module contains the actual file processor, which is the main object
+that carries out the conversion.
+"""
 import json
-import subprocess
 
-from ffconv import profiles
-
-
-def execute_cmd(cmd):
-    """
-    Wrapper around subprocess' Popen/communicate usage pattern, capturing
-    output and errors (which are raised).
-
-    :param cmd: shell command as string
-    :return: output of command
-    """
-    with subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT) as process:
-        output = b''
-        for line in iter(process.stdout.readline, b''):
-            if b'Error' in line:
-                raise ValueError(line.decode('utf-8'))
-            else:
-                output += line
-
-        retcode = process.poll()
-        if retcode:
-            raise subprocess.CalledProcessError(retcode, process.args, output=output)
-
-    # All good, decode output and return it
-    return output.decode('utf-8').lower()
-
-
-def log(msg, level=0):
-    parts = [' ' * (2 * level - 1), '- ' if level else '', msg]
-    print(''.join(parts))
+from . import profiles
+from .utils import execute_cmd, log
+from .stream_processors import VideoProcessor, AudioProcessor, SubtitleProcessor
 
 
 class FileProcessor(object):
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a docstring to this class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://localhost:9000/coding_rules#rule_key=python%3AS1720)